src/vs/workbench/browser/parts/menubar/menubarPart.ts

```diff
@@ -330,6 +330,13 @@ export class MenubarPart extends Part {
 				this.container.removeClass('inactive');
 			} else {
 				this.container.addClass('inactive');
+				if (!hasFocus) {
+					if (this.isVisible && this.currentMenubarVisibility === 'toggle') {
```

Comment:
Maybe extract this code to a reusable method and call it also from the `window.on("click")` code below? It seems here you check for the menu being visible but you are not doing that in the other place.